### PR TITLE
Fix challenging "boolean" usernames

### PR DESCRIPTION
**Pull request by DaWoblefet**
This particularly affected challenging user false, who would properly receive the challenge, but it would be canceled on the challenger's end later due to 'false' !== false in the later check on client-mainmenu:729. I'm not sure if there's a smarter way to do this.

We have more calls to data.('userid') - should I just go ahead and sanitize all of them?

Before:
image

After:
image

### CODE REVIEW

File: js/client-mainmenu.js [hunk 1]
```diff
@@ -727,6 +727,9 @@
 								$challenge.html('<form class="battleform"><p>The challenge was cancelled.</p><p class="buttonbar"><button name="dismissChallenge">OK</button></p></form>');
 							}
 						} else if ($challenge.find('button[name=cancelChallenge]').length && challengeToUserid !== userid) {
+							//console.log($challenge.find('button[name=cancelChallenge]').length);
+							//console.log(challengeToUserid);
+							//console.log(userid);
```
Comment (AnnikaCodes) [inline js/client-mainmenu.js:732]:
```suggestion
```

**Comment by Zarel** [comment]
Ugh, that's annoying. I think it'd be best if we stopped using jQuery's `data` entirely. I hate APIs that try to be "magic" because it leads to bugs like this.

File: js/client-mainmenu.js [hunk 2]
```diff
@@ -697,7 +697,7 @@
 			var self = this;
 			this.$('.pm-window').each(function (i, el) {
 				var $pmWindow = $(el);
-				var userid = $pmWindow.data('userid');
+				var userid = '' + $pmWindow.data('userid');
```
Comment (Zarel) [inline js/client-mainmenu.js:700]:
```suggestion
				var userid = el.dataset && el.dataset.userid;
```

**Comment by Zarel** [comment]
I'd suggest refactoring all the other `.data` calls into `getAttribute`, which should solve this problem.

**Comment by DaWoblefet** [comment]
I attempted to do a proper refactor and ended up breaking the client's functionality pretty badly. I've done a good deal of playtesting with the 'false' account and haven't noticed any other unusual behavior, so if a refactor should be done, that's beyond the scope of what I want to accomplish in this PR.

**Comment by Zarel** [comment]
Sure, we can leave it like this.

**Pull request merged**